python/tskit/trees.py

```diff
@@ -4549,6 +4549,62 @@ def subset(self, nodes, record_provenance=True):
         tables.subset(nodes, record_provenance)
         return tables.tree_sequence()
 
+    def union(
```

Comment:
I'm really not seeing the point of having a thin wrapper for this on the TreeSequence. It just means that we have to duplicate loads of documentation.

I say we just make this a TableCollection method, and only add a wrapper on the TreeSequence later if it's clear we need it.

Reply:
Also, if we haven't released ``subset``, I'd say we remove the TreeSequence method for that too. We're just making a bunch work for ourselves here, for the sake of the user typing ``ts.tables.subset()`` rather than ``ts.subset``.

Reply:
I kind of like the idea of having the method for both TableCollection and TreeSequence, but I see your point about it being redundant and confusing (doc-wise). what do you think @petrelharp? maybe @hyanwong could weigh in.

Reply:
I vote to remove the tree sequence method, as Jerome suggests. Tree sequences are immutable anyhow, so providing lots of methods that let you modify them isn't very natural (and could lead to more copying of tables than is necessary).

Reply:
exactly because they are immutable that typing `ts.tables.subset()` does not work. you'd have to dump the tables onto a new object (copying) and then perform the `subset` on it. for instance:
```
In [7]: ts = msprime.simulate(sample_size=2)

In [8]: print(ts.tables.nodes)
id	flags	population	individual	time	metadata
0	1	0	-1	0.00000000000000
1	1	0	-1	0.00000000000000
2	0	0	-1	0.26671842298487

In [9]: ts.tables.subset([0])

In [10]: print(ts.tables.nodes)
id	flags	population	individual	time	metadata
0	1	0	-1	0.00000000000000
1	1	0	-1	0.00000000000000
2	0	0	-1	0.26671842298487
```

Reply:
I think there's arguments either way. But, as @mufernando pointed out offline, many other tables methods have tree sequence methods as well. And, it makes sense for these to be tree sequence methods because they return valid tree sequence tables if they are given valid tree sequence tables, and are designed to work on valid tree sequence tables. So, I guess I'd say that if @mufernando wants to, then go for it.

Reply:
I don't care about it that much - it's up to you @mufernando.